specification/web/resource-manager/Microsoft.DomainRegistration/stable/2024-11-01/Domains.json

```diff
@@ -0,0 +1,1616 @@
+                  "ExpirationNotInRenewalTimeRange",
+                  "SubscriptionNotActive"
+                ],
+                "type": "string",
+                "x-ms-enum": {
+                  "name": "ResourceNotRenewableReason",
+                  "modelAsString": true
+                }
+              },
+              "readOnly": true
+            },
+            "dnsType": {
+              "description": "Current DNS type",
+              "enum": [
+                "AzureDns",
+                "DefaultDomainRegistrarDns"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "DnsType",
+                "modelAsString": false
+              }
+            },
+            "dnsZoneId": {
+              "description": "Azure DNS Zone to use",
+              "type": "string"
+            },
+            "targetDnsType": {
+              "description": "Target DNS type (would be used for migration)",
+              "enum": [
+                "AzureDns",
+                "DefaultDomainRegistrarDns"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "DnsType",
+                "modelAsString": false
+              }
+            },
+            "authCode": {
+              "type": "string",
+              "x-ms-mutability": [
+                "create",
+                "read"
+              ]
+            }
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "DomainPurchaseConsent": {
+      "description": "Domain purchase consent object, representing acceptance of applicable legal agreements.",
+      "type": "object",
+      "properties": {
+        "agreementKeys": {
+          "description": "List of applicable legal agreement keys. This list can be retrieved using ListLegalAgreements API under <code>TopLevelDomain</code> resource.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "agreedBy": {
+          "description": "Client IP address.",
+          "type": "string"
+        },
+        "agreedAt": {
+          "format": "date-time",
+          "description": "Timestamp when the agreements were accepted.",
+          "type": "string"
+        }
+      }
+    },
+    "DomainRecommendationSearchParameters": {
+      "description": "Domain recommendation search parameters.",
+      "type": "object",
+      "properties": {
+        "keywords": {
+          "description": "Keywords to be used for generating domain recommendations.",
+          "type": "string"
+        },
+        "maxDomainRecommendations": {
+          "format": "int32",
+          "description": "Maximum number of recommendations.",
+          "type": "integer"
+        }
+      }
+    },
+    "HostName": {
+      "description": "Details of a hostname derived from a domain.",
+      "type": "object",
+      "properties": {
+        "name": {
```

Comment:
>name [](http://example.com/codeflow?start=9&length=4)

Don't duplicate names of top-level ARM properties within the property bag. Could use hostName.